lib/cc/engine/analyzers/command_line_runner.rb

```diff
@@ -30,26 +50,17 @@ def run(input)
 
         attr_reader :command, :timeout
 
-        # Work around a race condition in JRuby's Open3.capture3 that can lead
-        # to a nil status returned. We'll consider the process successful if it
-        # produced output that can be parsed as JSON.
-        #
-        # https://github.com/jruby/jruby/blob/master/lib/ruby/stdlib/open3.rb#L200-L201
-        #
-        def handle_open3_race_condition(out)
-          JSON.parse(out)
-          NullStatus.new(true, 0)
+        def successful_output?(id)
+          output = File.read("/tmp/#{id}.out")
+          JSON.parse(output, max_nesting: false) and output
```

Comment:
This method is kind of a compound method, inspired by [this @pbrisbin comment](https://github.com/codeclimate/codeclimate-duplication/pull/138#discussion_r82686451), where it returns a useful truthy value when it's successful or a falsey value when it wasn't successful. In this case, I'm relying on `JSON.parse` to raise an error, so the `and` (or `&&`) is only used in the happy path, anyway. I'm tempted to change it to

```ruby
JSON.parse(output, max_nesting: false)
output
```

to make it more clear that we don't care about the truthiness of the first clause of the expression. WDYT?